packages/graphql_server/src/service/twitter-scraping-service.ts

```diff
@@ -0,0 +1,165 @@
+
+dotenv.config()
+
+type TwitterUser = {
+  profileName: string
+  isVerified: boolean
+  bio: string
+  followers: number
+  websiteUrl: string
+  handle: string
+  postsInfo: {
+    text: string
+    id: string
+    replies: number
+    retweets: number
+    likes: number
+    hashtags: string[]
+    post_url: string
+    owner_url: string
+    media_type: string
+    isRetweeted: boolean
+  }[]
+}
+
+type TwitterPost = {
+  id: string
+  text: string
+  retweetCount: number
+  likeCount: number
+  replies: number
+}
+
+type TwitterProfileResponse = {
+  profile_name: string
+  isVerified: boolean
+  bio: string
+  following: number
+  followers: number
+  website_url: string
+  posts: number
+  handle: string
+  collected_number_of_posts: number
+  posts_info: {
+    text: string
+    id: string
+    replies: number
+    retweets: number
+    likes: number
+    hashtags: string[]
+    post_url: string
+    owner_url: string
+    media_type: string
+    isRetweeted: boolean
+  }[]
+}
+
+const username = 'typescript2023'
+const apiKey = process.env.SCRAPING_BOT_API_KEY || ''
+const apiEndPoint = 'http://api.scraping-bot.io/scrape/data-scraper'
+const auth = 'Basic ' + Buffer.from(username + ':' + apiKey).toString('base64')
+
+const SORT_BY_RETWEETS = 'retweets'
+const SORT_BY_LIKES = 'likes'
+
+function sleep(ms: number): Promise<void> {
+  return new Promise((resolve) => {
+    setTimeout(resolve, ms)
+  })
+}
+
+/**
+ * Fetches the Twitter user by his handle
+ * @param twitterHandle
+ * @returns the TwitterUser with id, name and count of followers
+ */
+async function getTwitterUserByHandle(twitterHandle: string): Promise<TwitterUser> {
+  try {
+    // builds the axios request configuration
+    const requestConfig: AxiosRequestConfig = {
+      headers: {
+        Accept: 'application/json',
+        Authorization: auth
+      }
+    }
+    // specifies the used scraping mechanism, which is in this case twitterProfile
+    const requestData = {
+      scraper: 'twitterProfile',
+      url: 'https://twitter.com/' + twitterHandle
+    }
+
+    const response = await axios.post<{ responseId: string }>(
+      apiEndPoint,
+      requestData,
+      requestConfig
+    )
+
+    let data: TwitterProfileResponse | null = null
+
+    // sends the request until finalData is received
+    do {
+      await sleep(5000)
+      const responseUrl = `http://api.scraping-bot.io/scrape/data-scraper-response?scraper=twitterProfile&responseId=${response.data.responseId}`
+      const finalDataResponse = await axios.get<TwitterProfileResponse>(responseUrl, requestConfig)
+      data = finalDataResponse.data
+    } while (data === null)
+
+    return {
+      profileName: data.profile_name ?? '',
+      isVerified: data.isVerified ?? false,
+      bio: data.bio ?? '',
+      followers: data.following ?? -1,
+      websiteUrl: data.website_url ?? '',
+      handle: data.handle ?? '',
+      postsInfo: data.posts_info ?? []
+    }
+  } catch (error) {
```

Comment:
propagate errors to the caller, so we can give appropriate feedback in GraphQL resolver